frontend/src/container/AdminUserSpacesConfig.jsx

```diff
@@ -1,146 +1,173 @@
+          return props.workspaceList.find(space => space.id === userSpace.id) || fillMemberList(userSpace)
+        })).then((spaceListResult) => {
+          setSpaceList(sortWorkspaceList(spaceListResult))
+        })
         break
       }
       default: props.dispatch(newFlashMessage(props.t('Error while loading user')))
     }
   }
 
-  fillMemberList = async (space) => {
-    const fetchMemberList = await this.props.dispatch(getWorkspaceMemberList(space.workspace_id))
+  const handleMemberModified = (data) => {
+    setSpaceList(spaceList.map(space => {
+      if (space.id === data.fields.workspace.workspace_id) {
+        return {
+          ...space,
+          memberList: space.memberList.map(member => {
+            if (member.id === data.fields.user.user_id) {
+              return { ...member, ...serializeMember({ user: data.fields.user, ...data.fields.member }) }
+            } else {
+              return member
+            }
+          })
+        }
+      } else {
+        return space
+      }
+    }))
+  }
 
-    return {
-      ...space,
-      memberList: fetchMemberList.json || []
-    }
+  const handleMemberDeleted = async (data) => {
+    setSpaceList(spaceList.map(space => {
+      if (space.id === data.fields.workspace.workspace_id) {
+        return {
+          ...space,
+          memberList: space.memberList.filter(member => member.id !== data.fields.user.user_id)
+        }
+      } else {
+        return space
+      }
+    }))
   }
 
-  handleLeaveSpace = async (space) => {
-    const { props } = this
-    if (!space.workspace_id) return
+  const handleMemberCreated = async (data) => {
+    const space = spaceList.find(space => space.id === data.fields.workspace.workspace_id)
 
-    const fetchResult = await props.dispatch(deleteWorkspaceMember(space.workspace_id, props.userToEditId))
-    if (fetchResult.status !== 204) {
-      props.dispatch(newFlashMessage(props.t('Error while leaving the space'), 'warning'))
+    if (space.memberList) {
+      setSpaceList(spaceList.map(space => {
+        if (space.id === data.fields.workspace.workspace_id) {
+          return {
+            ...space,
+            memberList: [
+              ...space.memberList,
+              serializeMember({ user: data.fields.user, ...data.fields.member })
+            ]
+          }
+        } else {
+          return space
+        }
+      }))
+    } else {
+      setSpaceList(sortWorkspaceList([...spaceList, fillMemberList(space)]))
     }
   }
 
-  onlyManager = (member, memberList) => {
-    const manager = ROLE.workspaceManager.slug
+  const handleLeaveSpace = async (space) => {
+    if (!space.id) return
 
-    if (member.role !== manager) {
-      return false
+    const fetchResult = await props.dispatch(deleteWorkspaceMember(space.id, props.userToEditId))
+    if (fetchResult.status !== 204) {
+      props.dispatch(newFlashMessage(props.t('Error while leaving the space'), 'warning'))
     }
```

Comment:
A good reflex is to surround your await function by a try/catch, like:

```suggestion
    try {
        const fetchResult = await props.dispatch(deleteWorkspaceMember(space.id, props.userToEditId))
        if (fetchResult.status !== 204) {
            props.dispatch(newFlashMessage(props.t('Error while leaving the space'), 'warning'))
        }
    } catch (e) {
        console.log('Something when wrong when trying to leave the space.')
        console.log('Error:', e)
    }
```